modules/jdbc/src/main/java/org/testcontainers/jdbc/ConnectionUrl.java

```diff
@@ -0,0 +1,209 @@
+package org.testcontainers.jdbc;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.StringJoiner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+/**
+ * This is an Immutable class holding JDBC Connection Url and its parsed components, used by {@link ContainerDatabaseDriver}.
+ * <p>
+ * {@link ConnectionUrl#parseUrl()} method must be called after instantiating this class.
+ *
+ * @author manikmagar
+ */
+public class ConnectionUrl {
+
+    @Getter
+    private String url;
+
+    private String databaseType;
+
+    @Getter
+    private String imageTag = "latest";
+
+    private String dbHostString;
+
+    @Getter
+    private boolean inDaemonMode = false;
+
+    @Getter
+    private Optional<String> databaseHost = Optional.empty();
+
+    @Getter
+    private Optional<Integer> databasePort = Optional.empty();
+
+    @Getter
+    private Optional<String> databaseName = Optional.empty();
+
+    @Getter
+    private Optional<String> initScriptPath = Optional.empty();
+
+    @Getter
+    private Optional<InitFunctionDef> initFunction = Optional.empty();
+
+    @Getter
+    private Optional<String> queryString;
+
+    private ConnectionUrl() {
+        //Not Allowed here
+    }
+
+    public ConnectionUrl(final String url) {
+        this.url = Objects.requireNonNull(url, "Connection URL cannot be null");
+    }
+
+    public String getDatabaseType() {
+        return Objects.requireNonNull(this.databaseType, "Database Type cannot be null. Have you called parseUrl() method?");
+    }
+
+
+    /**
+     * This is a part of the connection string that may specify host:port/databasename.
+     * It may vary for different clients and so clients can parse it as needed.
+     *
+     * @return
+     */
+    public String getDbHostString() {
+        return Objects.requireNonNull(this.dbHostString, "Database Host String cannot be null. Have you called parseUrl() method?");
+    }
+
+    public static boolean accepts(final String url) {
+        return url.startsWith("jdbc:tc:");
+    }
+
+    public void parseUrl() {
```

Comment:
It looks like this is only used together with the constructor. Could we instead have a static factory method that instantiates, parses and returns?

Reply:
Makes sense. I will add a static method `newInstance(final String url)` that returns ready to use instance of ConnectionUrl.